### PR TITLE
Post Editor Header: Make block toolbar toggle button focus visible

### PR DESCRIPTION
**Pull request by t-hamano**
Part of #58293

## What?

This PR fixes an issue where the focus of the block tool toggle button is hidden when the fixed toolbar is enabled in the post editor.

### Before

image

### After

image

## Why?

Because the header toolbar has `overflow: hidden` applied for horizontal scrolling.

## How?

~~Add 2px padding to the right side of the header toolbar. This approach is already applied in the Site Editor:~~

https://github.com/WordPress/gutenberg/blob/fdf8fa62de27243ffa3179cd37ed6b39bdf141dc/packages/edit-site/src/components/header-edit-mode/style.scss#L27-L30

Update: I used the `--wp-admin-border-width-focus` variable in both the Site Editor and the Post Editor. This variable will have the appropriate value depending on the screen resolution.

## Testing Instructions

- Open a post.
- Enable Top Toolbar.
- Select a block.
- Narrow the browser width.
- Focus on the block tool toggle button.

### CODE REVIEW

File: packages/edit-post/src/components/header/style.scss [hunk 1]
```diff
@@ -25,6 +25,10 @@
 	// Leave enough room for the focus ring to show.
 	padding: 2px 0;
 	align-items: center;
+	// Allow focus ring to be fully visible on furthest right button.
+	@include break-medium() {
+		padding-right: 2px;
```
Comment (jasmussen) [inline packages/edit-post/src/components/header/style.scss]: Potentially you can use `$border-width-focus-fallback` here, sometimes using those SCSS variables simply make the code easier to read.

**Review by jasmussen — APPROVED** [review]
I can't deeply test this at the moment, have to finish some other things, but the code alone looks good. Thanks for the PR.

File: packages/edit-post/src/components/header/style.scss (hunk 1 above)
Comment (t-hamano) [reply packages/edit-post/src/components/header/style.scss]: Thanks for the review! 

In this case, it seemed better to use the `--wp-admin-border-width-focus` variable, so I updated it that way. This variable has an appropriate value depending on the screen resolution:

https://github.com/WordPress/gutenberg/blob/fdf8fa62de27243ffa3179cd37ed6b39bdf141dc/packages/base-styles/_mixins.scss#L455-L460

**Comment by t-hamano** [comment]
Sorry, I forgot to merge this PR 😅 I have imported the latest trunk into this PR, but it has been fixed as expected, so I would like to merge it.

**Pull request merged**